Remove serf dependency from ping/latency check

bernardjkim opened the pull request:
## Description
<!--Required. Provide high-level overview of what the change is for.-->
This PR removes serf dependency from the ping/latency check. Previously the ping check was using the serf coordinates API call to check latency between nodes. Now the latency check (renamed from ping check) pulls latency metrics from the nethealth service to check latency.

The nethealth service now exposes a latency summary for each peer. Latency is available at intervals of 10 percentiles as well as the 95th and 99th percentile. The latency is collected in milliseconds. Example output of the fetched latency metrics:
```
# HELP nethealth_echo_latency_summary_milli The round trip time between peers in milliseconds
# TYPE nethealth_echo_latency_summary_milli summary
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.1"} 18
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.2"} 25
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.3"} 47
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.4"} 47
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.5"} 59
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.6"} 81
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.7"} 81
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.8"} 81
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.9"} 87
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.95"} 87
nethealth_echo_latency_summary_milli{node_name="node-1",peer_name="node-2",quantile="0.99"} 87
nethealth_echo_latency_summary_milli_sum{node_name="node-1",peer_name="node-2"} 398
nethealth_echo_latency_summary_milli_count{node_name="node-1",peer_name="node-2"} 7
```

This updated implementation allows the latency check to recover more quickly by observing latency at a higher frequency and examining a smaller window of values.

This change is in preparation for completely removing serf dependency from satellite. 

## Linked tickets and PRs
* Refs https://github.com/gravitational/gravity/issues/2067

bernardjkim committed:
Remove serf dependency from ping/latency checker

- Add latency summaries to the set of exposed nethealth metrics.
- Add nethealth client interface.
- Move ping checker -> latency checker.
- Latency checker now checks latency using the exposed latency
summaries.

bernardjkim committed:
Remove ping checker

r0mant commented on cmd/agent/checkers.go:
```diff
@@ -79,11 +79,6 @@ func addToMaster(node agent.Agent, config *config, kubeConfig monitoring.KubeCon
 		return trace.Wrap(err)
 	}
 
-	pingHealth, err := monitoring.PingHealth(config.serfRPCAddr, config.serfMemberName)
```
Add the new latency checker instead of the removed ping checker here?

bernardjkim committed:
Address feedback

- Dial with context.
- socket -> socketPath
- metrics func returns parsed metrics
- Bug fix: Peroperly initialize summary vector

bernardjkim committed:
Remove HdrHistogram dependency

bernardjkim committed:
Fix label selector

bernardjkim committed:
Reduce latency summary MaxAge

bernardjkim committed:
Add latency checker

bernardjkim committed:
Remove incorrect comment
